providers/openshift/provider.go

```diff
@@ -643,3 +681,36 @@ func getKubeAPIURLWithPath(path string) *url.URL {
 
 	return ret
 }
+
+func (p *OpenShiftProvider) getAnonymousClient() (*rest.Config, error) {
```

Comment:
why not just expose https://github.com/deads2k/oauth-proxy/blob/60455f2656700d473797bce5315253cc5ceb3035/providers/openshift/authentication.go#L193-L194 and use that (or similar) during construction  It's in the same repo.

Reply:
that would've been clever, I added that exposed method to both the delegating* structs in the latest revision